compiler/qsc_partial_eval/src/lib.rs

```diff
@@ -967,6 +996,39 @@ impl<'a> PartialEvaluator<'a> {
         Value::unit()
     }
 
+    fn resolve_args(&self, store_pat_id: StorePatId, value: Value) -> Vec<Arg> {
+        let pat = self.package_store.get_pat(store_pat_id);
+        match &pat.kind {
+            PatKind::Discard => vec![Arg::Discard(value)],
```

Comment:
I actually think you can just panic on this case... a discard in a argument list is only used for creating a new lambda via partial application, which will have been lifted by earlier parts of compilation. That also means that `Arg::Discard` can probably go away, leaving `Arg` to become a tuple-struct with what today is in `Arg::Variable`. Perhaps something like `struct Arg(pub LocalVariableId, Variable);` would be sufficient.

Reply:
I think you can also have discards even in regular callables:
```qsharp
function Foo(_ : Int) : Int {
    0
}
```

Based on the HIR in the playground, the argument to `Foo` is a discard.